pkg/collector/providers/file.go

```diff
@@ -138,27 +158,30 @@ func collectDir(parentPath string, folder os.FileInfo) []check.Config {
 	subEntries, err := ioutil.ReadDir(dirPath)
 	if err != nil {
 		log.Warnf("Skipping config directory: %s", err)
-		return configs
+		return configs, defaultConfigs
 	}
 
 	// strip the trailing `.d`
-	checkName := folder.Name()[:len(folder.Name())-2]
+	checkName := folder.Name()[:len(folder.Name())-len(dirExt)]
 
 	// try to load any config file in it
 	for _, sEntry := range subEntries {
 		if !sEntry.IsDir() {
-			filePath := filepath.Join(dirPath, sEntry.Name())
-			conf, err := GetCheckConfigFromFile(checkName, filePath)
+			conf, isDefault, err := c.collectEntry(sEntry, dirPath, checkName)
 			if err != nil {
-				log.Warnf("%s is not a valid config file: %s", sEntry.Name(), err)
 				continue
```

Comment:
Could you add a comment `// logging already done in collectEntry` so it's not seen as an oversight?